homework/Truong_Duong/lesson_4/javascript_exercise.md

```diff
@@ -0,0 +1,203 @@
+# Javascript excercises
+
+**1. Choose the correct answer and explain your choice. What's the output?**
+
+```js
+function sayHi() {
+  console.log(name);
+  console.log(age);
+  var name = 'Lydia';
+  let age = 21;
+}
+
+sayHi();
+```
+
+A: Lydia and undefined
+B: Lydia and ReferenceError
+C: ReferenceError and 21
+D: undefined and ReferenceError
+
+> **Answer: D**
+> Explain: variable defined with `var` was hoisted the `declaration part` into the begining of the function, but the value did not. Therefore, it returned `undefined` with `name`
+> While with variable defined with `let`, a ES6 syntax, it wasn't hoisted by default. Therefore, it returned a ReferenceError here.
```

Comment:
I think when using `let` or `const` to declare variables, it still be hoisted but we should notice about the TDZ (temporal dead zone)
You can read more here: https://developer.mozilla.org/en-US/docs/Glossary/Hoisting?retiredLocale=vi#let_and_const_hoisting